src/config.rs

```diff
@@ -366,17 +382,20 @@ pub struct Config {
     pub display: DisplayConfig,
     pub updates: UpdatesConfig,
     pub directories: DirectoriesConfig,
+    pub config_file_path: PathWithSource,
 }
 
 impl Config {
     /// Convert a `RawConfig` to a high-level `Config`.
     ///
     /// For this, some values need to be converted to other types and some
     /// defaults need to be set (sometimes based on env variables).
-    fn from_raw(raw_config: RawConfig, relative_path_root: &Path) -> Result<Self> {
+    fn from_raw(raw_config: RawConfig, config_file_path: PathWithSource) -> Result<Self> {
         let style = raw_config.style.into();
         let display = raw_config.display.into();
         let updates = raw_config.updates.try_into()?;
+        // Safe to unwrap, it's a file path, so it should have a directory component
+        let relative_path_root = config_file_path.path().parent().unwrap();
```

Comment:
_should_! I am not so sure, let's not risk it and make this a `.context(...)?` instead (in particular for future-proofing)